src/libtorch_c/torch_c.cpp

```diff
@@ -338,10 +338,13 @@ extern "C" void* torchLoadModel(const char* graph, size_t graphlen, DLDeviceType
     ctx->cu = nullptr;
   }
   catch(std::exception& e) {
-    size_t len = strlen(e.what());
+    const char* err_message = e.what(); 
+    const size_t len = strlen(err_message);
     *error = (char*)alloc(len * sizeof(char));
-    strcpy(*error, e.what());
-    delete ctx;
+    if(*error){
+       strcpy(*error, err_message);
+    }
+    // delete ctx;
```

Comment:
We should probably still find a way to deallocate `ctx`

Reply:
Agree. But I would wait for some extra unit tests to make sure that the double freeing of the shared pointers is not only visible on ci